backend/src/apiserver/server/task_server.go

```diff
@@ -40,6 +41,19 @@ func (s *TaskServer) validateCreateTaskRequest(request *api.CreateTaskRequest) e
 	if task.GetPipelineName() == "" {
 		return errMustSpecify("PipelineName")
 	}
+	if task.GetNamespace() == "" {
+		return errMustSpecify("Namespace")
+	}
+	if strings.HasPrefix(task.GetPipelineName(), "namespace/") {
+		s := strings.Split(task.GetPipelineName(), "/")
+		if len(s) != 4 {
+			return util.NewInvalidInputError("invalid PipelineName for namespaced pipelines, need to follow 'namespace/${namespace}/pipeline/${pipelineName}': %s", task.GetPipelineName())
+		}
```

Comment:
note, we don't have any restrictions on pipeline name so far, so it may contain "/".
Recommend using https://pkg.go.dev/strings#SplitN instead.

Reply:
done